#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9999981656564052 -0.001451271142606431 -0.0012499983581111423 1.422052120190714
  position -0.35868822067665374 1.3868090770776251 0.03200679894848065
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2.1445 1.4855
  floorTileSize 0.33 0.33
}
DEF MAZE Group {
  children [
    DEF cornerStone Solid {
      translation -0.7424999999999818 0.032499426115 -0.41249999999998105
      rotation 0.05083534464876438 0.9987039688406258 -0.0024799912144156046 -5.965727021567361e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(4)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation 0.7425000000000183 0.032499426115 -0.41249999999998105
      rotation 0.05083534464876438 0.9987039688406258 -0.0024799912144156046 -5.965727021567361e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(5)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation 0.7425000000000183 0.032499426115 0.4125000000000189
      rotation 0.05083534464876438 0.9987039688406258 -0.0024799912144156046 -5.965727021567361e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(6)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation -0.7424999999999818 0.032499426115 0.4125000000000189
      rotation 0.05083534464876438 0.9987039688406258 -0.0024799912144156046 -5.965727021567361e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(7)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF hWall Solid {
      translation 6.200081795896482e-13 0.032444141860000004 -0.41249999977157836
      rotation -4.8675996231186774e-08 0.9999999999999988 -7.871406398847454e-11 1.189117794812066e-08
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      name "solid(1)"
      boundingObject USE hWall
      physics Physics {
        density 100
      }
    }
    DEF hWall Solid {
      translation 6.200081795896482e-13 0.032444141860000004 0.4125000002284216
      rotation -4.8675996231186774e-08 0.9999999999999988 -7.871406398847454e-11 1.189117794812066e-08
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      boundingObject USE hWall
      physics Physics {
        density 100
      }
    }
    DEF vWall Solid {
      translation -0.7424999999263409 0.03246939280000001 4.0750874211419483e-13
      rotation 1.4533531880094549e-09 0.9999999999999996 2.6307721024573534e-08 -6.782941681214416e-09
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(2)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
    }
    DEF vWall Solid {
      translation 0.7425000000736592 0.03246939280000001 4.0750874211419483e-13
      rotation 1.4533531880094549e-09 0.9999999999999996 2.6307721024573534e-08 -6.782941681214416e-09
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(3)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
    }
    DEF Pole MazeBlock {
      translation -0.9049999999999252 -1.1477699999942748e-06 -0.5774999999999251
      rotation -0.05197696085381925 0.9044269068576044 0.42345054692654127 -1.4703809217750323e-16
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.577499999999925 -1.1477699999942748e-06 -0.24749999999992497
      rotation 0.21964059640383454 0.9296854103425505 -0.29570770400442986 2.766379239700383e-16
      name "maze block(20)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999247 -1.1477700000012137e-06 -0.24749999999992492
      rotation -0.39406202190951733 0.7532447443901288 0.5266284059346142 -5.92412453846143e-16
      name "maze block(21)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992486 -1.1477699999942748e-06 -0.247499999999925
      rotation 0.2090236914137296 0.9755176288318567 -0.06836996611122352 4.4385736896119454e-16
      name "maze block(22)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992502 -1.1477699999942748e-06 -0.24749999999992503
      rotation -0.8386841908138469 0.48928949163504926 0.23917487630275458 -2.5275063968274055e-16
      name "maze block(23)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007508 -1.1477699999942748e-06 -0.24749999999992492
      rotation 0.09132702566306374 0.9759962017738038 -0.19771390569873074 -9.45603917555219e-16
      name "maze block(25)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007525 -1.1477700000012137e-06 -0.24749999999992492
      rotation -0.39406202190951733 0.7532447443901288 0.5266284059346142 -5.92412453846143e-16
      name "maze block(26)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999251 -1.1477699999942748e-06 -0.08249999999992494
      rotation 0.014905169191076938 0.9996314991872777 -0.022687039559652646 -1.2925261910120023e-15
      name "maze block(30)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999246 -1.1477700000012137e-06 -0.08249999999992497
      rotation 0.5099575903645435 0.5187193572136081 -0.6862022183594932 -7.362552108621045e-16
      name "maze block(31)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.2474999999999251 -1.1477700000012137e-06 -0.08249999999992494
      rotation 0.557903074246801 0.7587910885577154 -0.3361253392283391 -6.2718846922825365e-16
      name "maze block(32)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992494 -1.1477700000012137e-06 -0.08249999999992501
      rotation -0.6758537530324775 0.7259672121164583 -0.1272529427705785 -7.760091048638022e-16
      name "maze block(33)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007507 -1.1477700000012137e-06 -0.08249999999992501
      rotation -0.6758537530324775 0.7259672121164583 -0.1272529427705785 -7.760091048638022e-16
      name "maze block(34)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007513 -1.1477700000012137e-06 -0.08249999999992504
      rotation -0.2132707864601891 0.730264819290527 -0.6490214675565259 6.508555857556607e-16
      name "maze block(35)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007536 -1.1477700000012137e-06 -0.08249999999992497
      rotation 0.5099575903645435 0.5187193572136081 -0.6862022183594932 -7.362552108621045e-16
      name "maze block(36)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.577500000000075 -1.1477699999942748e-06 -0.08249999999992494
      rotation 0.014905169191076938 0.9996314991872777 -0.022687039559652646 -1.2925261910120023e-15
      name "maze block(37)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999251 -1.1477699999942748e-06 0.08250000000007507
      rotation 0.014905169191076938 0.9996314991872777 -0.022687039559652646 -1.2925261910120023e-15
      name "maze block(40)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999246 -1.1477700000012137e-06 0.08250000000007504
      rotation 0.5099575903645435 0.5187193572136081 -0.6862022183594932 -7.362552108621045e-16
      name "maze block(41)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.2474999999999251 -1.1477700000012137e-06 0.08250000000007507
      rotation 0.557903074246801 0.7587910885577154 -0.3361253392283391 -6.2718846922825365e-16
      name "maze block(42)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992494 -1.1477700000012137e-06 0.082500000000075
      rotation -0.6758537530324775 0.7259672121164583 -0.1272529427705785 -7.760091048638022e-16
      name "maze block(43)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007507 -1.1477700000012137e-06 0.082500000000075
      rotation -0.6758537530324775 0.7259672121164583 -0.1272529427705785 -7.760091048638022e-16
      name "maze block(44)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007513 -1.1477700000012137e-06 0.08250000000007497
      rotation -0.2132707864601891 0.730264819290527 -0.6490214675565259 6.508555857556607e-16
      name "maze block(45)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007536 -1.1477700000012137e-06 0.08250000000007504
      rotation 0.5099575903645435 0.5187193572136081 -0.6862022183594932 -7.362552108621045e-16
      name "maze block(46)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.577499999999925 -1.1477699999942748e-06 0.24750000000007508
      rotation 0.017916537696845335 0.9994675760085618 -0.02726833666593803 8.917362264920179e-16
      name "maze block(50)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999246 -1.1477700000012137e-06 0.247500000000075
      rotation 0.20543748728787545 0.9732749176759562 0.10262248018687153 7.008745402077594e-16
      name "maze block(51)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992506 -1.1477699999942748e-06 0.24750000000007497
      rotation -0.46567011186836654 0.841767265855121 0.2730919571267103 7.559074401732286e-17
      name "maze block(52)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992494 -1.1477700000012137e-06 0.24750000000007505
      rotation -0.6770400705142299 0.5419617724029964 -0.49788972691941463 3.013383004812751e-16
      name "maze block(53)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007507 -1.1477700000012137e-06 0.24750000000007505
      rotation -0.6770400705142299 0.5419617724029964 -0.49788972691941463 3.013383004812751e-16
      name "maze block(54)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007497 -1.1477699999942748e-06 0.24750000000007502
      rotation 0.009323920295736565 0.9998602732341244 0.0138743835358854 1.0217313991130293e-15
      name "maze block(55)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007536 -1.1477700000012137e-06 0.247500000000075
      rotation 0.20543748728787545 0.9732749176759562 0.10262248018687153 7.008745402077594e-16
      name "maze block(56)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.5775000000000751 -1.1477699999942748e-06 0.24750000000007508
      rotation 0.017916537696845335 0.9994675760085618 -0.02726833666593803 8.917362264920179e-16
      name "maze block(57)"
      length 0.015
    }
    DEF hBlock MazeBlock {
      translation -0.6599999999996644 -1.071251999999745e-05 -0.5774999999905034
      rotation -1.904518011585281e-08 0.9999999999999996 -1.0336926841872377e-08 4.797605970888842e-09
      name "maze block(10)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.3300000000003351 -1.071251999999745e-05 0.08250000000949669
      rotation 2.820323148696261e-08 0.9999999999999988 -3.8344317670692884e-08 4.797606860504783e-09
      name "maze block(62)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.1650000000003353 -1.0712520000004389e-05 -0.08249999999050339
      rotation -2.4703756834642294e-08 0.9999999999999997 -7.634803995648463e-09 4.797606064518601e-09
      name "maze block(61)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.1650000000003353 -1.0712520000004389e-05 0.08250000000949662
      rotation -2.4703756834642294e-08 0.9999999999999997 -7.634803995648463e-09 4.797606064518601e-09
      name "maze block(60)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.3300000000003351 -1.071251999999745e-05 -0.24749999999050315
      rotation 1.4297311351616327e-08 0.9999999999999998 1.4092726411880102e-09 4.797606509187662e-09
      name "maze block(59)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.4950000000003351 -1.071251999999745e-05 -0.08249999999050331
      rotation -4.359594553025057e-08 0.9999999999999992 5.381511848906639e-09 4.797606441873213e-09
      name "maze block(58)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.495000000000335 -1.071251999999745e-05 0.2475000000094968
      rotation 4.078814083121113e-08 0.9999999999999992 1.6112405466801484e-08 4.797606319038322e-09
      name "maze block(39)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.16500000000033527 -1.071251999999745e-05 0.2475000000094966
      rotation 2.151849672441326e-08 0.9999999999999998 -8.835832921347292e-09 4.797606210455966e-09
      name "maze block(29)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.9049999999905034 -1.0712520000004389e-05 -0.3300000000003353
      rotation -3.544875197392676e-18 1 5.045678541130297e-18 1.5707996986971802
      name "maze block(78)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.41249999999050296 -1.071251999999745e-05 -3.3531530544856427e-13
      rotation 5.2101607053930933e-17 1 -4.4470034568005154e-17 1.5707996986971806
      name "maze block(65)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.2475000000094968 -1.071251999999745e-05 0.16499999999966483
      rotation 1.5855418688747785e-16 1 -1.3503004761124175e-16 1.5707996986971806
      name "maze block(64)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.24749999999050343 -1.0712520000004389e-05 -3.352913959682593e-13
      rotation 1.2415382419142653e-16 1 -1.5235376021715613e-17 1.5707996986971806
      name "maze block(63)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.2475000000094967 -1.071251999999745e-05 -0.16500000000033527
      rotation 4.107683670785605e-17 1 -1.9338076959501364e-17 1.570799698697181
      name "maze block(49)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125000000094968 -1.071251999999745e-05 -0.16500000000033524
      rotation 9.31220942321653e-17 1 4.2966095850310476e-17 1.5707996986971802
      name "maze block(11)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125000000094968 -1.071251999999745e-05 0.3299999999996648
      rotation 2.1168217763770656e-16 1 -1.7454100571421044e-16 1.5707996986971806
      name "maze block(48)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.494999999999665 -1.071251999999745e-05 -0.24749999999050343
      rotation -1.8407229792124206e-08 0.9999999999999998 -2.7599041535091285e-09 4.797605991195431e-09
      name "maze block(4)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.6599999999996644 -1.071251999999745e-05 -0.0824999999905033
      rotation 4.9805586203240755e-08 0.9999999999999987 -1.5140514510256e-08 4.7976062434500534e-09
      name "maze block(6)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 3.3527146936788123e-13 -1.0712520000004389e-05 -0.0824999999905032
      rotation 1.4669938506403902e-08 0.9999999999999993 3.000440061495204e-08 4.7976065049251364e-09
      name "maze block(8)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 3.3527146936788123e-13 -1.0712520000004389e-05 0.08250000000949681
      rotation 1.4669938506403902e-08 0.9999999999999993 3.000440061495204e-08 4.7976065049251364e-09
      name "maze block(9)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.32999999999966495 -1.071251999999745e-05 0.08250000000949669
      rotation 2.820323148696261e-08 0.9999999999999988 -3.8344317670692884e-08 4.797606860504783e-09
      name "maze block(12)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.3299999999996649 -1.071251999999745e-05 0.24750000000949676
      rotation 7.688498838440146e-09 0.9999999999999997 -2.9369175096183092e-08 4.797606029484609e-09
      name "maze block(13)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050332 -1.071251999999745e-05 0.3299999999996648
      rotation -1.5686772802342625e-17 1 6.948009293492973e-18 1.5707996986971802
      name "maze block(18)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050332 -1.071251999999745e-05 -0.33000000000033525
      rotation -1.5686772802342625e-17 1 6.948009293492973e-18 1.5707996986971802
      name "maze block(2)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.0824999999905032 -1.0712520000004389e-05 -3.3532856612273895e-13
      rotation -1.957263012401386e-16 1 2.1177157230932275e-16 1.570799698697181
      name "maze block(17)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.24749999999050323 -1.0712520000004389e-05 -0.16500000000033524
      rotation 1.4913847753331904e-16 1 -1.0179121527266343e-16 1.570799698697181
      name "maze block(15)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5774999999905036 -1.071251999999745e-05 0.3299999999996648
      rotation -5.2936820799673264e-17 1 1.200451449109678e-16 1.5707996986971806
      name "maze block(1)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5774999999905038 -1.0712520000004389e-05 0.1649999999996648
      rotation 5.0848187824630175e-17 1 -8.041674840406232e-17 1.5707996986971806
      name "maze block(3)"
      length 0.14
    }
  ]
}
DEF z5207653 E-puck {
  translation -0.66 0 -0.33
  rotation -2.8600689825558306e-07 0.9999999936607209 -0.00011259874165793287 -3.1378736895997212
  controller "z5207653_MTRN4110_PhaseA"
  camera_width 320
  camera_height 240
  turretSlot [
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 1.309
      name "ds0"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 1.8326
      name "ds1"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 2.8798
      name "ds2"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 -2.8797853071795863
      name "ds3"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 -1.8325953071795862
      name "ds4"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 -1.3089953071795861
      name "ds5"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 -0.261795307179586
      name "ds6"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DEF ds DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 0.261805
      name "ds7"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 1.5707996938995747
      name "DS_FRONT"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DistanceSensor {
      translation 0 0 -0.01
      name "DS_RIGHT"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
    DistanceSensor {
      translation 0 0 -0.01
      rotation 0 1 0 -3.141592653589793
      name "DS_LEFT"
      lookupTable [
        0 0 0
        0.15 1500 0
      ]
    }
  ]
  groundSensorsSlot [
    Camera {
      translation -0.01 0 0
      rotation -0.577349935856137 -0.577349935856137 0.5773509358560258 2.09439
      name "groundCam"
      recognitionColors [
        0 0 0
      ]
      width 128
      height 128
      exposure 6
    }
    DistanceSensor {
      name "ground 1"
      type "infra-red"
      numberOfRays 5
    }
  ]
}
